pkg/cni/cilium.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cni
+
+import (
+	"encoding/json"
+	"fmt"
+
+	appskubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/apps.kubermatic/v1"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	ciliumAppDefName    = "cilium"
+	ciliumHelmChartName = "cilium"
+)
+
+// CiliumApplicationDefinitionCreator creates Cilium ApplicationDefinition managed by KKP to be used
+// for installing Cilium CNI into KKP usr clusters.
+func CiliumApplicationDefinitionCreator() reconciling.NamedAppsKubermaticV1ApplicationDefinitionCreatorGetter {
+	return func() (string, reconciling.AppsKubermaticV1ApplicationDefinitionCreator) {
+		return ciliumAppDefName, func(app *appskubermaticv1.ApplicationDefinition) (*appskubermaticv1.ApplicationDefinition, error) {
```

Comment:
we can use CNIPluginTypeCiliuminstead of creating a new constant with the same value?